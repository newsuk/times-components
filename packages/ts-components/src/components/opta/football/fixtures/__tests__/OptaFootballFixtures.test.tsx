import React from 'react';
import { render, waitForElementToBeRemoved } from '@testing-library/react';

import 'regenerator-runtime';
import '@testing-library/jest-dom';

jest.mock('@times-components/image', () => ({
  Placeholder: () => <>Placeholder</>
}));

const mockInitSettings = jest.fn();
const mockInitStyleSheet = jest.fn();
const mockInitComponent = jest.fn();

const mockInitElement = () => {
  const element = document.createElement('div');
  element.appendChild(document.createTextNode('Widget'));
  return element;
};

jest.mock('../../../utils/config', () => ({
  initSettings: mockInitSettings,
  initStyleSheet: mockInitStyleSheet,
  initScript: () => new Promise(resolve => resolve({})),
  initElement: mockInitElement,
  initComponent: mockInitComponent
}));

import { OptaFootballFixtures } from '../OptaFootballFixtures';

const requiredProps = {
  season: '2020',
  competition: '3',
  date_from: '2021-06-20',
  date_to: '2021-07-11'
};

describe('OptaFootballFixtures', () => {
  it('should render correctly', async () => {
    const { asFragment, getByText } = render(
      <OptaFootballFixtures {...requiredProps} />
    );
    expect(asFragment()).toMatchSnapshot();

    await waitForElementToBeRemoved(getByText('Placeholder'));

    expect(mockInitSettings).toHaveBeenCalledTimes(1);
    expect(mockInitStyleSheet).toHaveBeenCalledTimes(1);
    expect(mockInitComponent).toHaveBeenCalledTimes(1);

    expect(asFragment()).toMatchSnapshot();
  });
});
