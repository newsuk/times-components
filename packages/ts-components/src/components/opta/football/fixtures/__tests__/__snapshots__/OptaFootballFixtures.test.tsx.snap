// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptaFootballFixtures should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa giUBsz"
  >
    <div
      class="sc-htpNat sc-ifAKCX ceXVkP"
    />
    <div
      class="sc-bwzfXH kVYHKf"
    >
      Placeholder
    </div>
  </div>
</DocumentFragment>
`;

exports[`OptaFootballFixtures should render correctly 2`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa hGXeaj"
  >
    <div
      class="sc-htpNat sc-ifAKCX ceXVkP"
    >
      <div>
        Widget
      </div>
    </div>
  </div>
</DocumentFragment>
`;
